### PR TITLE
Require PHP 7.2.9+, up from 5.3+

### PR DESCRIPTION
**Pull request by jdforrester**

### CODE REVIEW

**Comment by jdforrester** [comment]
With this, I'll cut 1.8.2, which we'll actually upgrade MW to (MW is still running 1.8.0, whoops).

File: composer.json [hunk 1]
```diff
@@ -18,10 +18,10 @@
 		}
 	],
 	"require": {
-		"PHP" : ">=5.3"
+		"PHP" : ">=7.2.9"
 	},
 	"require-dev": {
-		"phpunit/phpunit": "~4.8.36 || 6.5.14 || 7.5.14"
+		"phpunit/phpunit": "6.5.14 || 7.5.14"
```
Comment (Krinkle) [inline composer.json]: Can drop 6.5 as well, given we already have code for 7.5 here.

Comment (jdforrester) [reply composer.json]: Yeah, I was thinking we might want to drop 6.x and 7.x and jump to 8.x.

**Comment by franzliedke** [comment]
Hello and thanks for keeping this library maintained!

I would urge you to reconsider making this (very much breaking) change in a patch release. Now that the original library has been abandoned, we (the [Flarum](https://flarum.org)) would like to switch to the WIkimedia fork. For example, because the fork supports PHP 7.4. We currently still support PHP 7.1, though - and now, with the 1.8.2 release, we can have either 7.4 support or 7.1 support, but not both. :cry: 

Since the changelog claims you have switched to Semantic Versioning, I would argue that this change breaks that promise.

Of course, tagging a new major release would solve that problem. :wink:

**Comment by jdforrester** [comment]
Hey, yes, sorry, the next release wil reflect this breaking change.

Note that PHP 7.1 stopped getting bug fixes [in December 2018](https://www.php.net/supported-versions.php) and stopped getting security fixes in December 2019.

**Comment by chrisdeeming** [comment]
Sorry to drag this up after so long.

@jdforrester can you recall why the minimum became PHP 7.2.9 rather than any earlier version of PHP 7.2? I've commonly seen PHP 7.2.5 be used as a minimum by some packages.

Was this just the tested version at the time or were there significant issues prior to PHP 7.2.9?

We're looking into including wikimedia/less.php in XenForo. We currently have our own fork of `oyejorge/less.php` which we maintain here: https://github.com/xenforo-ltd/less.php. We may need to PR some changes into this release.

We were aiming for PHP 7.2 as a minimum but this package has the highest requirements yet so trying to rationalise that 🙂

Many thanks!

**Comment by jdforrester** [comment]
Hey! Sorry for the slow response.

We run into a bunch of bugs with PHP itself in running Wikipedia, and we're moderately successful in getting them fixed upstream; when that happens, we bump to the new version and set that as our standard level. I don't know off-hand what segfaults or other bugs specifically led us to require 7.2.*9* however, sorry (I know that for PHP 7.3.x we require 7.3.19+ due to CVE-2019-11048, but we didn't start listing causes until more recently.

Nowadays we require 7.2.22, [due to a segfault](https://phabricator.wikimedia.org/T228346), but we've not yet made this breaking change upgrade demand in this library (or others).

(Note that PHP 7.2 is EOL, but we're still using it in Wikimedia production thanks to Debian's extended security support.)

Hope this helps!

**Pull request merged**